mysql/data.go

```diff
@@ -126,9 +126,19 @@ func convScalar(spannerType ddl.Type, srcTypeName string, TimezoneOffset string,
 	}
 }
 
-func convBool(val string) (bool, error) {
+func convBool(conv *internal.Conv, val string) (bool, error) {
 	b, err := strconv.ParseBool(val)
 	if err != nil {
+		// MySQL allows non-boolean values in boolean columns.
+		// If we can't parse as boolean, try to re-parse as INT64
+		// and treat as true if value is non-zero.
+		// Note: if ParseBool fails, then val is probably a non-zero number.
+		i, err2 := convInt64(val)
```

Comment:
we should consider adding a test for convBool since the behavior is slightly non-trivial now

Reply:
Done.
Added tests for this change and also for other bool case we didn't test before.
Also added a negative test e.g. that 128 fails with an error.